lib/matplotlib/axes/_axes.py

```diff
@@ -5112,6 +5176,15 @@ def reduce_C_function(C: array) -> float
 
             if zscale == "log":
                 bin_edges = np.geomspace(zmin, zmax, nbins + 1)
+            elif zscale == "symlog":
+                zmin, zmax = symlog_transform[zname].transform_non_affine(
+                    np.array([zmin, zmax])
+                )
+                bin_edges = np.linspace(zmin, zmax, nbins + 1)
+                bin_edges = inv_symlog_trans[zname].transform_non_affine(
+                    bin_edges
+                )
```

Comment:
@timhoffm introducing `symlog_transform` and `inv_symlog_trans` as dict seemed logic to me when considering this for loop. I can obviously get rid of it but I thought it would be better not to. What's your take on this?